app/src/main/java/ca/unb/mobiledev/reflexrevolution/instructions/SwipeInstruction.java

```diff
@@ -12,6 +12,13 @@
     private final TouchDetector touchDetector;
     private TouchDetector.SwipeAction currentAction;
     private Random rand;
+    private Random rand2;
+    private int index;
+    private String[] rightOptions = {"RIGHT", "OPPOSITE OF LEFT"};
+    private String[] leftOptions = {"LEFT", "OPPOSITE OF RIGHT"};
+    private String[] upOptions = {"UP", "OPPOSITE OF DOWN"};
+    private String[] downOptions = {"DOWN", "OPPOSITE OF UP"};
```

Comment:
This works well. An possible improvement we could do is to merge all of it into 1 data structure instead of 4 arrays by using a HashMap. Something like this:
### Define HashMap
```java
private final Map<TouchDetector.SwipeAction, String[]> textLabels = new HashMap<>();
```
### Initialize values in `setup()`
Each array is associated with the enum value directly.
```java
// Add labels
textLabels.put(TouchDetector.SwipeAction.SWIPE_RIGHT, new String[] {"RIGHT", "OPPOSITE OF LEFT"});
textLabels.put(TouchDetector.SwipeAction.SWIPE_LEFT, new String[] {"LEFT", "OPPOSITE OF RIGHT"});
textLabels.put(TouchDetector.SwipeAction.SWIPE_UP, new String[] {"UP", "OPPOSITE OF DOWN"});
textLabels.put(TouchDetector.SwipeAction.SWIPE_DOWN, new String[] {"DOWN", "OPPOSITE OF UP"});
```
### Get value to display
Don't need a switch case anymore!
```java
@Override
public void display() {
    if (currentAction.getType() == TouchDetector.Type.SWIPE) {
        // Add small label for swipe instructions
        int index = rand.nextInt(2);
        String displayTxt = Objects.requireNonNull(textLabels.get(currentAction))[index];
        addTextView(displayTxt, SMALL_TEXT_SIZE);
        // Main label
        addTextView("SWIPE", DEFAULT_TEXT_SIZE);
    }
    else {
        // Main label
        addTextView("FLICK", DEFAULT_TEXT_SIZE);
    }
}
```